### PR TITLE
Not run link handler registration in test execution

### PR DESCRIPTION
**Pull request by BeckerWdf**

### CODE REVIEW

**Comment by BeckerWdf** [comment]
This PR depends on https://github.com/eclipse-platform/eclipse.platform.ui/pull/2302

**Comment by BeckerWdf** [comment]
@jukzi: Are the log entries in the test execution now gone for this PR?

**Comment by jukzi** [comment]
> @jukzi: Are the log entries in the test execution now gone for this PR?

i am not sure what you asking.
https://download.eclipse.org/eclipse/downloads/drops4/I20240923-1800/testresults/ep434I-unit-mac64-java17_macosx.cocoa.x86_64_17/org.eclipse.debug.tests.AutomatedSuite.txt still shows thousends. 

```
java.lang.RuntimeException: Some job is still running or waiting to run: 
'AutoRegisterSchemeHandlersJob(19)'/org.eclipse.urischeme.AutoRegisterSchemeHandlersJob:RUNNING
```
but its only in the i-build not tested for every PR

**Comment by BeckerWdf** [comment]
> but its only in the i-build not tested for every PR

I thought it's also visible in PR-builds. So we have to merge this and have a look into the next i-build

File: bundles/org.eclipse.ui.ide.application/src/org/eclipse/ui/internal/ide/application/JUnitTestUtil.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	public static boolean isJunitTestRunning() {
+		if (cachedIsJunitTestRunning == null) {
+			try {
+				if (Platform.isRunning()) {
+					AtomicBoolean result = new AtomicBoolean();
+					cachedIsJunitTestRunning = ServiceCaller.callOnce(JUnitTestUtil.class, EnvironmentInfo.class, envInfo -> {
+						String application = envInfo.getProperty("eclipse.application"); //$NON-NLS-1$
+						result.set(application != null && Set.of( //
+								// see org.eclipse.pde.internal.launching.IPDEConstants
+								"org.eclipse.pde.junit.runtime.nonuithreadtestapplication", // //$NON-NLS-1$
```
Comment (BeckerWdf) [inline bundles/org.eclipse.ui.ide.application/src/org/eclipse/ui/internal/ide/application/JUnitTestUtil.java:35]: @sratz: from the i-build's log 
https://download.eclipse.org/eclipse/downloads/drops4/I20240923-1800/testresults/ep434I-unit-mac64-java17_macosx.cocoa.x86_64_17/org.eclipse.debug.tests.AutomatedSuite.txt
i see 
`Framework arguments:  -application org.eclipse.test.uitestapplication formatter=...`
So it looks like `org.eclipse.test.uitestapplication`  is used. Is it missing in this list?

Comment (sratz) [reply bundles/org.eclipse.ui.ide.application/src/org/eclipse/ui/internal/ide/application/JUnitTestUtil.java:35]: Yes, I checked for all the test runners we have in platform / PDE / Tycho and these 3 are missing in the list:

- org.eclipse.pde.junit.runtime.coretestapplicationnonmain
- org.eclipse.test.uitestapplication
- org.eclipse.test.coretestapplication

**Review by sratz — APPROVED** [review]
lgtm, test failure is unrelated

**Comment by BeckerWdf** [comment]
> AutoRegisterSchemeHandlersJob

In the meantime I don't see them any more.
@jukzi: Can you confirm this is fixed?

**Comment by jukzi** [comment]
confirmed:
https://download.eclipse.org/eclipse/downloads/drops4/I20241007-1800/testresults/ep434I-unit-mac64-java17_macosx.cocoa.x86_64_17/org.eclipse.debug.tests.AutomatedSuite.txt
does not contain "AutoRegisterSchemeHandlersJob" anymore

**Pull request merged**